zkevm-circuits/src/state_circuit/state.rs

```diff
@@ -605,379 +456,119 @@ impl<
-            let target = if need_pad_start_row && i == start_offset {
-                START_TAG
-            } else {
-                target
-            };
-            region.assign_fixed(|| "target", self.q_target, i, || Ok(F::from(target as u64)))?;
-            region.assign_advice(|| "padding", self.padding, i, || Ok(F::one()))?;
-            region.assign_advice(|| "memory", self.flag, i, || Ok(F::one()))?;
-        }
-
-        Ok(())
-    }
-
     /// Assign cells.
     pub(crate) fn assign(
         &self,
         mut layouter: impl Layouter<F>,
         randomness: F,
-        memory_ops: Vec<Rw>,
-        stack_ops: Vec<Rw>,
-        storage_ops: Vec<Rw>,
-    ) -> Result<Vec<BusMapping<F>>, Error> {
-        let mut bus_mappings: Vec<BusMapping<F>> = Vec::new();
-
-        let address_diff_is_zero_chip = IsZeroChip::construct(self.address_diff_is_zero.clone());
-
-        let memory_address_monotone_chip =
-            MonotoneChip::<F, MEMORY_ADDRESS_MAX, true, false>::construct(
-                self.address_monotone.clone(),
-            );
-        memory_address_monotone_chip.load(&mut layouter)?;
-
-        let padding_monotone_chip =
-            MonotoneChip::<F, 1, true, false>::construct(self.padding_monotone.clone());
-        padding_monotone_chip.load(&mut layouter)?;
-
-        let storage_key_diff_is_zero_chip =
-            IsZeroChip::construct(self.storage_key_diff_is_zero.clone());
+        rw_map: &RwMap,
+    ) -> Result<(), Error> {
+        let key_is_same_with_prev_chips: [IsZeroChip<F>; 5] = [0, 1, 2, 3, 4]
+            .map(|idx| IsZeroChip::construct(self.key_is_same_with_prev[idx].clone()));
 
         layouter.assign_region(
             || "State operations",
             |mut region| {
-                let mut offset = 0;
-
-                let memory_mappings = self.assign_memory_ops(
-                    &mut region,
-                    randomness,
-                    memory_ops.clone(),
-                    &address_diff_is_zero_chip,
-                    offset,
-                );
-                bus_mappings.extend(memory_mappings.unwrap());
-                offset += MEMORY_ROWS_MAX;
-
-                let stack_mappings = self.assign_stack_ops(
-                    &mut region,
-                    randomness,
-                    stack_ops.clone(),
-                    &address_diff_is_zero_chip,
-                    offset,
-                );
-                bus_mappings.extend(stack_mappings.unwrap());
-                offset += STACK_ROWS_MAX;
-
-                let storage_mappings = self.assign_storage_ops(
-                    &mut region,
-                    randomness,
-                    storage_ops.clone(),
-                    &address_diff_is_zero_chip,
-                    &storage_key_diff_is_zero_chip,
-                    offset,
-                );
-                bus_mappings.extend(storage_mappings.unwrap());
+                let mut offset = 1;
```

Comment:
Could you explain why this starts with `offset = 1`?  Where is `offset = 0` set? (which should use the `START_TAG`).
In case you're planning on doing this in a future PR, would you add a small note with a TODO stating that please?